services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipInvitationManagerImpl.java

```diff
@@ -76,6 +77,10 @@ public MembershipInvtnSubmission create(UserInfo userInfo,
 		if (!authorizationManager.canAccess(
 				userInfo, mis.getTeamId(), ObjectType.TEAM, ACCESS_TYPE.TEAM_MEMBERSHIP_UPDATE).getAuthorized()) 
 			throw new UnauthorizedException("Cannot create membership invitation.");
+		Team team = teamDAO.get(mis.getTeamId());
+		if (team.getCanPublicJoin() != null && team.getCanPublicJoin()) {
+			throw new IllegalArgumentException("Cannot invite to public team.");
```

Comment:
This message will be shown to user. We need to provide more friendly and helpful message. For example, explaining that the team they are trying to invite users to is already open for public to join, therefore, a membership invitation is not needed here.